src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php

```diff
@@ -0,0 +1,207 @@
+            PgwPunchoutCatalogConnectionTableMap::COL_IS_ACTIVE,
+            PgwPunchoutCatalogConnectionTableMap::COL_FORMAT,
+            PgwPunchoutCatalogConnectionTableMap::COL_CREATED_AT,
+        ]);
+
+        $config->setSearchable([
+            PgwPunchoutCatalogConnectionTableMap::COL_ID_PUNCHOUT_CATALOG_CONNECTION,
+            PgwPunchoutCatalogConnectionTableMap::COL_NAME,
+            SpyCompanyTableMap::COL_NAME,
+        ]);
+
+        $config->setRawColumns([
+            static::COL_STATUS,
+            static::COL_FORMAT,
+            static::COL_ACTIONS,
+        ]);
+
+        return $config;
+    }
+
+    /**
+     * @param \Spryker\Zed\Gui\Communication\Table\TableConfiguration $config
+     *
+     * @return array
+     */
+    protected function prepareData(TableConfiguration $config)
+    {
+        /** @var \Propel\Runtime\Collection\ObjectCollection|\Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection[] $punchoutCatalogConnections */
+        $punchoutCatalogConnectionCollection = $this->runQuery(
+            $this->prepareQuery($this->connectionPropelQuery),
+            $config,
+            true
+        );
+
+        if ($punchoutCatalogConnectionCollection->count() === 0) {
+            return [];
+        }
+
+        return $this->mapPunchoutCatalogConnections($punchoutCatalogConnectionCollection);
+    }
+
+    /**
+     * @module Company
+     *
+     * @param \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionQuery
+     *
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionQuery
+     */
+    protected function prepareQuery(PgwPunchoutCatalogConnectionQuery $connectionPropelQuery): PgwPunchoutCatalogConnectionQuery
+    {
+        return $connectionPropelQuery->joinWithCompanyBusinessUnit()
+            ->useCompanyBusinessUnitQuery()
+            ->joinCompany()
+                ->withColumn(SpyCompanyTableMap::COL_NAME, static::COL_COMPANY)
+            ->endUse();
+    }
+
+    /**
+     * @param \Propel\Runtime\Collection\ObjectCollection|\Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection[] $punchoutCatalogConnectionCollection
+     *
+     * @return array
+     */
+    protected function mapPunchoutCatalogConnections(ObjectCollection $punchoutCatalogConnectionCollection): array
+    {
+        $punchoutCatalogConnectionRows = [];
+
+        foreach ($punchoutCatalogConnectionCollection as $punchoutCatalogConnection) {
+            $punchoutCatalogConnectionRows[] = $this->mapPunchoutCatalogConnection($punchoutCatalogConnection);
+        }
+
+        return $punchoutCatalogConnectionRows;
+    }
+
+    /**
+     * @param \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection $punchoutCatalogConnection
+     *
+     * @return array
+     */
+    protected function mapPunchoutCatalogConnection(PgwPunchoutCatalogConnection $punchoutCatalogConnection): array
```

Comment:
Maybe:
```
    protected function mapPunchoutCatalogConnection(PgwPunchoutCatalogConnection $punchoutCatalogConnection): array
    {
        $punchoutCatalogConnectionRow = $punchoutCatalogConnection->toArray();

        return array_merge($punchoutCatalogConnectionRow, [
            static::COL_ACTIONS => $this->buildLinks($punchoutCatalogConnection),
            static::COL_STATUS => $this->getStatusLabel($punchoutCatalogConnection),
            static::COL_FORMAT => $this->generateLabel(
                $punchoutCatalogConnection->getFormat(),
                'label-secondary'
            ),
            static::COL_CREATED_AT => $this->utilDateTimeService->formatDateTime(
                $punchoutCatalogConnection->getCreatedAt()
            ),
            static::COL_COMPANY => $punchoutCatalogConnection->getCompanyBusinessUnit()
                ->getCompany()
                ->getName(),
        ]);
    }
```

Reply:
It doesn't make any difference IMO

Reply:
@asmarovydlo From a functional perspective - maybe. But it arrays merge looks better for me than a lot of single operations.

Reply:
Would prefer to keep it than just to save some time :)
If there is not major difference